### PR TITLE
Refactoring of the code

### PR DESCRIPTION
**Pull request by Domefemia**
I have rewrote the code in a more pythonic way, change string interpolation with f string, removed unused library/variable but there more unused variable that i have not remove!

### CODE REVIEW

File: usr/lib/sticky/common.py [hunk 1]
```diff
@@ -11,7 +11,8 @@
 CONFIG_PATH = os.path.join(CONFIG_DIR, 'notes.json')
 SAVE_DELAY = 3
 
-backup_file_name = re.compile(r"\Abackup-[0-9]{10,}\.json$", re.IGNORECASE)
+backup_file_name = re.compile(r"\Abackup-[0-9]{10,}\.json$", {re.IGNORECASE}")
```
Comment (mtwebster) [inline usr/lib/sticky/common.py:14]: syntax errors with the curlied re.IGNORECASE and extra "

File: usr/lib/sticky/sticky.py [hunk 2]
```diff
@@ -9,16 +15,8 @@
 gi.require_version('Gspell', '1')
 gi.require_version('Gtk', '3.0')
 gi.require_version('XApp', '1.0')
```
Comment (mtwebster) [inline usr/lib/sticky/sticky.py:17]: These require_version calls *MUST* occur before the actual imports of those modules

**Comment by mtwebster** [comment]
Please have a look at my comments.

In light of those issues, I'm concerned over a potential lack of testing here. This is a pretty broad swath of changes - they're positive ones, for sure, but they can create new regressions, and care must be taken to perform enough testing to check out all possible code paths to changes here.

Also, please squash things into one or two logical commits. As I mentioned in one of your other PRs, something like one for imports, anothere for syntax isn't a bad idea.

**Comment by collinss** [comment]
Separate pull requests would also be a viable option. TBH, part of the reason I haven't gotten to this yet is because there's a lot to review in this one pull request, but if each unrelated change is in it's own pr, it's a lot easier to manage, and potential changes can be easier to address as well (just glancing over a couple of your changes I can see things I'd like changed further before merging).

File: usr/lib/sticky/common.py [hunk 3]
```diff
@@ -259,7 +254,7 @@ def flush(self):
 
     def new_group(self, group_name):
         if group_name in self.notes_lists:
-            if not confirm(_("Overwrite Existing Group"), _("There is already a group named '%s'. This action will overwrite it. Continue anyway?") % group_name, self.window):
+            if not confirm(_("Overwrite Existing Group"), _(f"There is already a group named '{group_name}'. This action will overwrite it. Continue anyway?{self.window}")):
```
Comment (mtwebster) [inline usr/lib/sticky/common.py:257]: Any strings marked for translation (`_(...)`)will need to stay in the traditional format, as these changes will break the existing translations, and I'm not even sure gettext supports f string format in the first place.

**Comment by mtwebster** [comment]
I don't know that it needs separate PRs, just a smaller group of commits - one for each 'type' of changes would be fine, but it's up to you.

**Pull request closed without merging**